src/QtImageReader.cpp

```diff
@@ -51,9 +60,70 @@ void QtImageReader::Open()
 	// Open reader if not already open
 	if (!is_open)
 	{
+
+#if USE_RESVG == 1
+		image = std::shared_ptr<QImage>(new QImage());
+		bool success;
+
+		QFileInfo image_file_info(QString::fromStdString(path));
+
+		// Only use resvg for files with a '.svg' extension
+		if (image_file_info.suffix() == "svg") {
+			// Init resvg structs
+			resvg_render_tree *rtree = NULL;
+			resvg_options opt;
+			// Init resvg options
+			opt.dpi = 144;
+			opt.draw_background = 0;
+			opt.fit_to.type = RESVG_FIT_TO_ORIGINAL;
+
+			// Load the svg data from the file into a byte array
+			QFile image_file(QString::fromStdString(path));
+			image_file.open(QFile::ReadOnly);
+			QByteArray ba = image_file.readAll();
+			ZmqLogger::Instance()->AppendDebugMethod("QtImageReader::Open", "ba.size()", ba.size(), "", -1, "", -1, "", -1, "", -1, "", -1);
+
+			// use byte array as data source for resvg
+			char *error = nullptr;
+			rtree = resvg_parse_rtree_from_data(ba.constData(), &opt, &error);
+			if (!rtree) {
+				ZmqLogger::Instance()->AppendDebugMethod("QtImageReader::Open", "resvg_parse_rtree_from_data error", -1, error, -1, "", -1, "", -1, "", -1, "", -1);
+				resvg_error_msg_destroy(error);
+				abort();
+			}
+
+			// Play a little game with resvg_size and resvg_rect due to disparate types in resvg_get_image_size() and resvg_qt_render_to_canvas()
+			resvg_size resvgSize;
+			resvg_rect resvgRect;
+			resvg_get_image_size(rtree, &resvgRect.width, &resvgRect.height);
+			resvgSize.width = resvgRect.width;
+			resvgSize.height = resvgRect.height;
+			ZmqLogger::Instance()->AppendDebugMethod("QtImageReader::Open", "resvgRect.width", resvgRect.width, "resvgRect.height", resvgRect.height, "resvgSize.width", resvgSize.width, "resvgSize.height", resvgSize.height, "", -1, "", -1);
+
+			// Render and save the svg image to a canvas which has a QImage object as the output
+			QImage img = QImage(resvgSize.width, resvgSize.height, QImage::Format_ARGB32_Premultiplied);
+			img.fill(Qt::transparent);
+			QPainter p;
+			p.begin(&img);
+			p.setRenderHint(QPainter::Antialiasing);
+			resvg_qt_render_to_canvas(rtree, &opt, resvgSize, &p);
+			p.end();
+
+			// Copy the QImage so it is usable below
+			image = std::shared_ptr<QImage>(new QImage(img));
```

Comment:
std::make_shared<QImage>(img);

Reply:
Is the suggestion that `(img)` is more efficient than `(new QImage(img))`?

Reply:
Ever so slightly, yes. Saves a copy. Outside a tight loop, doesn't mean too much, but still a best practice

Reply:
Gotcha!  I've updated the source, tested it and committed an update.